scripts/test/ISISPowderRunDetailsTest.py

```diff
@@ -0,0 +1,127 @@
+        self.assertEqual(output_obj.vanadium_run_numbers, output_obj.run_number)
+        self.assertEqual(expected_vanadium_runs, output_obj.output_run_string)
+
+    def test_callable_params_are_used(self):
+        # These attributes are based on a custom YAML file at the specified path
+        expected_label = "16_4"
+        expected_vanadium_runs = "11-12"
+        expected_empty_runs = "13-14"
+        expected_grouping_file_name = "grouping_file"
+        run_number_string = "17-18"
+        mock_inst = self.setup_mock_inst_settings(yaml_file_path="ISISPowderRunDetailsTestCallable.yaml")
+
+        # Get the YAML file as a dict first
+        wrapped_funcs = RunDetails.WrappedFunctionsRunDetails
+
+        yaml_callable = RunDetails.RunDetailsFuncWrapper(function=wrapped_funcs.get_cal_mapping_dict,
+                                                         run_number_string=run_number_string, inst_settings=mock_inst)
+
+        empty_callable = yaml_callable.add_to_func_chain(function=wrapped_funcs.cal_dictionary_key_helper,
+                                                         key="custom_empty_run_numbers")
+        vanadium_callable = yaml_callable.add_to_func_chain(function=wrapped_funcs.cal_dictionary_key_helper,
+                                                            key="custom_vanadium_run_numbers")
+        grouping_callable = RunDetails.RunDetailsFuncWrapper(function=lambda: expected_grouping_file_name)
+
+        output_obj = RunDetails.create_run_details_object(run_number_string=run_number_string, inst_settings=mock_inst,
+                                                          is_vanadium_run=True, empty_run_call=empty_callable,
+                                                          vanadium_run_call=vanadium_callable,
+                                                          grouping_file_name_call=grouping_callable)
+
+        self.assertEqual(output_obj.label, expected_label)
+        self.assertEqual(output_obj.empty_runs, expected_empty_runs)
+        self.assertEqual(output_obj.grouping_file_path,
+                         os.path.join(mock_inst.calibration_dir, expected_grouping_file_name))
+        self.assertEqual(output_obj.vanadium_run_numbers, expected_vanadium_runs)
+        self.assertEqual(output_obj.run_number, expected_vanadium_runs)
+
+    def test_run_details_splined_name_list_is_used(self):
+        expected_vanadium_runs = "11-12"
+        expected_offset_file_name = "offset_file_name"
+        splined_name_list = ["bar", "bang", "baz"]
+        mock_inst = self.setup_mock_inst_settings(yaml_file_path="ISISPowderRunDetailsTest.yaml")
+        output_obj = RunDetails.create_run_details_object(run_number_string=10, inst_settings=mock_inst,
+                                                          is_vanadium_run=False, splined_name_list=splined_name_list)
+
+        expected_splined_out_str = ''.join('_' + val for val in splined_name_list)
```

Comment:
Adding `[ ]` to make it a list -> **[**"_" + val for val in splined_name_list **]** is about 2x faster
```
In [62]: timeit('"".join(["_" + val for val in splined_name_list])',setup='splined_name_list = ["name","2","fff"]')
Out[62]: 0.4237639904022217
```
```
In [63]: timeit('"".join("_" + val for val in splined_name_list)',setup='splined_name_list = ["name","2","fff"]')
Out[63]: 0.9818799495697021
```